client/components/domains/register-domain-step/index.jsx

```diff
@@ -272,8 +280,14 @@ var RegisterDomainStep = React.createClass( {
 					if ( ! domain.match( /.{3,}\..{2,}/ ) || domain.match( /\.wordpress\.com/i ) ) {
 						return callback();
 					}
-
+					const timestamp = Date.now();
 					canRegister( domain, ( error, result ) => {
+						const timeDiff = Date.now() - timestamp;
+						if ( error && error.code ) {
```

Comment:
If I'm not mistaken this could be be simplified (and deduplicated - oh, delicious keywords ;)) to something like:
```
this.recordEvent( 'domainAvailabilityReceive', domain, ( error && error.code ) || 'available', timeDiff, this.props.analyticsSection );
```